x-pack/plugins/security_solution/public/common/components/sourcerer/index.tsx

```diff
@@ -128,18 +158,101 @@ export const Sourcerer = React.memo<SourcererComponentProps>(({ scope: scopeId }
 
   const handleSaveIndices = useCallback(() => {
     const patterns = selectedOptions.map((so) => so.label);
-    onChangeDataView(dataViewId, patterns);
+    if (dataViewId != null) {
+      onChangeDataView(dataViewId, patterns);
+    }
     setPopoverIsOpen(false);
   }, [onChangeDataView, dataViewId, selectedOptions]);
 
   const handleClosePopOver = useCallback(() => {
     setPopoverIsOpen(false);
     setExpandAdvancedOptions(false);
   }, []);
+
+  // deprecated timeline index pattern handlers
+  const onContinueUpdateDeprecated = useCallback(() => {
+    setIsShowingUpdateModal(false);
+    const patterns = selectedPatterns.filter((pattern) =>
+      defaultDataView.patternList.includes(pattern)
+    );
+    onChangeDataView(defaultDataView.id, patterns);
+    setPopoverIsOpen(false);
+  }, [defaultDataView.id, defaultDataView.patternList, onChangeDataView, selectedPatterns]);
+
+  const onUpdateDeprecated = useCallback(() => {
+    // are all the patterns in the default?
+    if (missingPatterns.length === 0) {
+      onContinueUpdateDeprecated();
+    } else {
+      // open modal
+      setIsShowingUpdateModal(true);
+    }
+  }, [missingPatterns, onContinueUpdateDeprecated]);
+
+  const [isTriggerDisabled, setIsTriggerDisabled] = useState(false);
+  const onOpenAndReset = useCallback(() => {
+    setPopoverIsOpen(true);
+    resetDataSources();
+  }, [resetDataSources]);
+  const onUpdateUiSettings = useCallback(async () => {
+    const defaultIndex = await uiSettings.get<string[]>(DEFAULT_INDEX_KEY);
+    const uiSettingsIndexPattern = [...defaultIndex, ...missingPatterns];
+    const isUiSettingsSuccess = await uiSettings.set(
+      DEFAULT_INDEX_KEY,
+      ensurePatternFormat(uiSettingsIndexPattern)
+    );
+
+    setIsShowingUpdateModal(false);
+    setPopoverIsOpen(false);
+
+    if (isUiSettingsSuccess) {
+      onChangeDataView(
+        defaultDataView.id,
+        // to be at this stage, activePatterns is defined, the ?? selectedPatterns is to make TS happy
+        activePatterns ?? selectedPatterns,
+        false
+      );
+      addSuccess({
+        color: 'success',
+        title: toMountPoint(i18n.SUCCESS_TOAST_TITLE),
+        text: toMountPoint(<RefreshButton />),
+        iconType: undefined,
+        toastLifeTimeMs: 600000,
+      });
+      setMissingPatterns([]);
+      setIsTriggerDisabled(true);
+    } else {
+      addError(new Error(i18n.FAILURE_TOAST_TITLE), {
+        title: i18n.FAILURE_TOAST_TITLE,
+        toastMessage: (
+          <>
+            <FormattedMessage
+              id="xpack.securitySolution.indexPatterns.failureToastText"
+              defaultMessage="Unexpected error occurred on update. If you would like to modify your data, you can manually select a data view {link}."
+              values={{
+                link: <EuiLink onClick={onOpenAndReset}>{i18n.TOGGLE_TO_NEW_SOURCERER}</EuiLink>,
+              }}
+            />
+          </>
+        ) as unknown as string,
```

Comment:
`TS2352: Conversion of type 'Element' to type 'string' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.`

Reply:
but i need the link here... maybe ill submit a ticket to core asking this prop be updated